File: Fugazi/src/org/fugazi/ql/ast/expression/literal/INT.java
package org.fugazi.ql.ast.expression.literal;

import org.fugazi.ql.ast.expression.IExpressionVisitor;
import org.fugazi.ql.ast.type.IntType;
import org.fugazi.ql.ast.type.Type;

public class INT extends Literal {

    private final int value;

    public INT(int _value) {
        super();

        this.value = _value;
    }
    public INT(int _value, int _lineNum) {
        super(_lineNum);

        this.value = _value;
    }

    public int getValue() {
        return this.value;
    }

    @Override
    public String toString() {
        return String.valueOf(this.value);
    }

    @Override
    public Type getReturnedType() {
        return new IntType();
    }

    public <T> T accept(IExpressionVisitor<T> visitor) {
        return visitor.visitINT(this);
    }
}

File: FelipezConde/QuestionnaireLanguage/AST/Nodes/Literals/Int.cs

namespace AST.Nodes.Literals
{
    public class Int : Literal
    {
        private readonly int value;

        public Int(int value, PositionInText positionInText)
            :base(positionInText)
        {
            this.value = value;
        }

        public int GetValue()
        {
            return value;
        }

        public override string ToString()
        {
            return "int";
        }

        public override Types.Type RetrieveType()
        {
            return new Types.IntType();
        }

        public override T Accept<T>(VisitorInterfaces.IExpressionVisitor<T> visitor)
        {
            return visitor.Visit(this);
        }
    }
}
